wallet-new/src/Cardano/Wallet/Kernel/Submission.hs

```diff
@@ -0,0 +1,431 @@
+    where
+        newEmptyState :: WalletSubmissionState
+        newEmptyState = WalletSubmissionState {
+              _wssPendingSet  = emptyPending
+            , _wssCurrentSlot = Slot 0
+            , _wssSchedule   = Schedule IntMap.empty mempty
+            }
+
+-- | Informs the 'WalletSubmission' layer about new 'Pending' transactions.
+addPending :: Pending -> WalletSubmission m -> WalletSubmission m
+addPending newPending ws =
+    let ws' = ws & over (wsState . wssPendingSet) (unionPending newPending)
+    in schedulePending newPending ws'
+
+-- | Removes the input 'Pending' from the local 'WalletSubmission' pending set.
+remPending :: Pending -> WalletSubmission m -> WalletSubmission m
+remPending updatedPending ws =
+    ws & over (wsState . wssPendingSet) (flip differencePending updatedPending)
+
+-- | A \"tick\" of the scheduler.
+-- Returns the set transactions which needs to be droppped by the system as
+-- they likely exceeded the submission count and they have no chance to be
+-- adopted in a block.
+-- @N.B.@ The returned 'WalletSubmission' does not come with an already-pruned
+-- local 'Pending' set. It will be caller's responsibility to call 'remPending'
+-- on the 'WalletSubmission'.
+--
+tick :: MonadThrow m
+     => WalletSubmission m
+     -- ^ The current 'WalletSubmission'.
+     -> m (Evicted, WalletSubmission m)
+     -- ^ The set of transactions upper layers will need to drop, together
+     -- with the new 'WalletSubmission'.
+tick ws = do
+    let wss               = ws  ^. wsState
+        currentSlot       = wss ^. wssCurrentSlot
+        rho               = _wsResubmissionFunction ws
+    case dueThisSlot currentSlot ws of
+         Left e -> throwM e
+         Right (dueNow, newSchedule) -> do
+            (toPrune, schedule') <- rho currentSlot dueNow newSchedule
+            let newState = ws & over (wsState . wssSchedule) (const schedule')
+                              . over (wsState . wssCurrentSlot) (mapSlot succ)
+            return (toPrune , newState)
+
+data SchedulingError =
+    LoopDetected Pending
+    -- ^ The transactions in this 'Pending' set forms a cycle and they
+    -- couldn't be top-sorted.
+
+instance Exception SchedulingError
+
+instance Show SchedulingError where
+    show (LoopDetected pending) = toString $ sformat ("LoopDetected " % F.build) pending
+
+-- | Returns a set of 'Pending' transactions which are due in the given
+-- 'Slot'.
+-- TODO(adn) Document this.
+--
+dueThisSlot :: Slot
+            -- ^ The current 'Slot'.
+            -> WalletSubmission m
+            -- ^ The 'WalletSubmissionState'.
+            -> Either SchedulingError ([ScheduledTx], Schedule)
+            -- ^ An error if no schedule can be produced, or the scheduled
+            -- transactions together with the new, updated 'Schedule'.
+dueThisSlot currentSlot ws =
+    let localPending = ws ^. wsState . wssPendingSet
+        (dependent, independent) = partitionScheduledTx localPending candidates
```

Comment:
Yes, fair point -- maybe `sendable` or some more specific name is more appropriate? We could surely add a disclaimer under the form of a comment if you think it's appropriate, in case we want to keep the `dependent vs independent` partition going.

Reply:
A middle ground option might be naming them (canSend, cannotSend) ..